### ISSUE TITLE
Game closes after completing Fisherman mission and exiting dialogue

### ISSUE THREAD

**Issue by Alterforce** [CleverRaven/Cataclysm-DDA#49524]
# Describe the bug

Upon giving exiting dialogue after completing the mission, the game automatically closes.

# Steps To Reproduce

1. Find Fisherman NPC
2. Complete quests until you get the "find nails" mission
3. Give nails to Fisherman NPC
4. Exit dialogue

# Expected behavior

Game does not automatically close upon exiting dialogue after completing the mission.

# Versions and configuration

 - OS: Windows 10
 - Game Version: 68b8751
 - Graphics version:  Tiles
 - Ingame language: English
 - Mods loaded: dda, no_npc_food, package_bionic_professions

# Additional context

Game does not automatically close when giving Jack Isherwood 10 3L jars for "Find 10 3L jars"  #mission.

Save file:
[La Veta.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/6753996/La.Veta.zip)

**Comment by ZhilkinSerg** [comment]
What is in the `crash.log` and/or `debug.log`?

**Comment by Alterforce** [comment]
> What is in the `crash.log` and/or `debug.log`?
[debug for La Veta.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/6757735/debug.for.La.Veta.zip)

After reproducing the bug, I suspect that the boat I made has something to do with the closing of the game.
If it is in the same map tile as the NPC when the dialogue is exited then the game closes.
I tested it twice without the boat, once by teleporting with the debug menu and another by swimming to the NPC's location.
The boat is made of 3 wooden frames, 3 wooden boat hulls, 3 wooden seats, and 1 hand paddles.
